:sparkles: Add timestamp, id and checksum info for a received message

lsabi opened the pull request:
Add metadata for received message.

I'm not sure if it makes sense to also add:
- `MessageState`
- `Headers`
- `Length`

I think that for now, `timestamp` and `id` are the most important, especially if we consider the implementation of the `Pollingstrategy` with the timestamp (which I'm trying to use). Since `checksum` is also a primitive data type, I've also added it (why not?)

mmodzelewski commented on src/receive_message.rs:
```diff
@@ -35,6 +35,27 @@ impl ReceiveMessage {
+    /// Retrieves the checksum of the received message.
+    ///
+    /// The checksum represents the integrity of the message within its topic.
+    pub fn checksum(&self) -> u64 {
```
please change the return type to u32, to match the checksum from the inner PolledMessage

mmodzelewski commented:
> I'm not sure if it makes sense to also add:
> 
> * `MessageState`
> * `Headers`
> * `Length`

The end goal would be to have full capabilities of the underlying SDK, so I'd say we should include these eventually.

lsabi commented:
@mmodzelewski I've fixed the `checksum`'s return and added also the remaining information, but the `headers` which I don't know what is a good way of representing them. Maybe a `dict`? If you have any suggestions, let me know or feel free to edit and then merge the branch.

mmodzelewski commented on src/receive_message.rs:
```diff
@@ -35,6 +36,46 @@ impl ReceiveMessage {
+    /// State represents the state of the response.
+    pub fn state(&self) -> String {
+        match self.inner.state {
+            RustMessageState::Available => "Available".to_string(),
```
@lsabi maybe you could add an enum for these values instead of mapping to string?

lsabi replied:
There's also the actual name in the `Display`. We could return a `string` but formatted the same way?

[https://github.com/iggy-rs/iggy/blob/master/sdk/src/models/messages.rs#L101](https://github.com/iggy-rs/iggy/blob/master/sdk/src/models/messages.rs#L101)

mmodzelewski commented:
I think `dict` would be good for headers, but it will require more work as it can hold different types of data. We can leave it for another PR.

lsabi commented:
@mmodzelewski did you get a chance to look at the PR?

lsabi commented:
@mmodzelewski have looked at this PR?

mmodzelewski commented:
Hey @lsabi, I added a commit with an enum implementation I mentioned in an earlier comment.
